full-service/src/service/decorated_types.rs

```diff
@@ -284,3 +286,92 @@ pub struct JsonProof {
     pub txo_id: String,
     pub proof: String,
 }
+
+#[derive(Deserialize, Serialize, Default, Debug)]
+pub struct StringifiedJsonUnspentTxOut {
+    pub tx_out: JsonTxOut,
+    pub subaddress_index: String,
+    pub key_image: String,
+    pub value: String,
+    pub attempted_spend_height: String,
+    pub attempted_spend_tombstone: String,
+    pub monitor_id: String,
+}
+
+impl From<&JsonUnspentTxOut> for StringifiedJsonUnspentTxOut {
+    fn from(src: &JsonUnspentTxOut) -> Self {
+        Self {
+            tx_out: src.tx_out.clone(),
+            subaddress_index: src.subaddress_index.to_string(),
+            key_image: src.key_image.clone(),
+            value: src.value.clone(),
+            attempted_spend_height: src.attempted_spend_height.to_string(),
+            attempted_spend_tombstone: src.attempted_spend_tombstone.to_string(),
+            monitor_id: src.monitor_id.clone(),
+        }
+    }
+}
+
+impl From<&StringifiedJsonUnspentTxOut> for JsonUnspentTxOut {
+    fn from(src: &StringifiedJsonUnspentTxOut) -> Self {
+        Self {
+            tx_out: src.tx_out.clone(),
+            subaddress_index: src.subaddress_index.parse::<u64>().unwrap(),
+            key_image: src.key_image.clone(),
+            value: src.value.clone(),
+            attempted_spend_height: src.attempted_spend_height.parse::<u64>().unwrap(),
+            attempted_spend_tombstone: src.attempted_spend_tombstone.parse::<u64>().unwrap(),
+            monitor_id: src.monitor_id.clone(),
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Default, Debug)]
+pub struct StringifiedJsonTxProposal {
+    pub input_list: Vec<StringifiedJsonUnspentTxOut>,
+    pub outlay_list: Vec<JsonOutlay>,
+    pub tx: JsonTx,
+    pub fee: String,
+    pub outlay_index_to_tx_out_index: Vec<(String, String)>,
```

Comment:
Why do these need to be strings? They will never exceed a very small number.

Reply:
I think we need to keep it consistent - they are u64 in the tx_proposal object, and we are categorically saying that all u64s are strings in json

Reply:
Fair, although I'd argue that maybe we should change them to u16 or u32 since there's no world in which there are that many outlays/txouts...

Reply:
Fair will fix up in [719](https://github.com/mobilecoinfoundation/mobilecoin/pull/719)

Reply:
Done!